src/main/java/team/themoment/hellogsmv3/domain/application/controller/ApplicationController.java

```diff
@@ -21,11 +19,26 @@ public class ApplicationController {
 
     private final AuthenticatedUserManager manager;
     private final CreateApplicationService createApplicationService;
+    private final ModifyApplicationService modifyApplicationService;
 
     @PostMapping("/application/me")
     public ResponseEntity<Map<String, String>> create(@RequestBody @Valid ApplicationReqDto dto) {
         createApplicationService.execute(dto, manager.getId());
         return ResponseEntity.status(HttpStatus.CREATED).body(Map.of("message", "생성되었습니다"));
     }
 
+    @PutMapping("/application/me")
+    public ResponseEntity<Map<String, String>> modify(@RequestBody @Valid ApplicationReqDto dto) {
+        modifyApplicationService.execute(dto, manager.getId(), false);
+        return ResponseEntity.status(HttpStatus.OK).body(Map.of("message", "수정되었습니다"));
```

Comment:
```suggestion
        return ResponseEntity.status(HttpStatus.OK).body(CommonApiResponse.success());
```
와 같이 공통 객체를 통해 관리하는게 안정성에 도움이 될 듯 합니다.

Reply:
controller 딴에서 매번 ResponseEntity를 생성하여 리턴하는게 아니라.
controller 에서는 `ApiResponse` 에 대한 객체만 던지고, controllerAdvice 같은 장치에서 global 하게 `ApiResponse` 를 관리하는 방법도 고려 부탁드립니다.

Reply:
@jyeonjyan 

```java
return ResponseEntity.status(HttpStatus.OK).body(ApiResponse.success("생성되었습니다."));
```
```java
return ApiResponse.success("생성되었습니다.");
```

이렇게 ResponseEntity로 감싸서 반환하는 방식에서 반환 객체만 던지는 방식을 다른 프로젝트에서 많이 보아서 이런식으로 구현하는 것에 대해 고민한 적이 있습니다.

API의 반환값은 객체만 던지고, 예외가 발생하면 controllerAdvice에서 핸들링하는 방식으로 구현될 것같은데 이렇게 된다면 201이나 204같은 상태코드를 반환하지 못하고 200의 상태코드만 반환되는것 처럼 보입니다.

다른 2xx대의 상태코드를 사용하지않고 성공시에는 200의 상태코드만 반환하고 예외만 핸들링하여 따로 처리해주는 방식이 더 좋은지 궁금합니다.

Reply:
세 개 다 너무 좋은 질문입니다 ! 

* 이렇게 ResponseEntity로 감싸서 반환하는 방식에서 반환 객체만 던지는 방식을 다른 프로젝트에서 많이 보아서 이런식으로 구현하는 것에 대해 고민한 적이 있습니다.
  * **👍🏻 ApiResponse 는 springboot 에서 제공하는 클래스인것 같은데요. 
  hellogsm 내부에서 통신하는 api 들의 commonApiResponse 를 정의하고 맞게 클래스를 생성해보면 좋을 것 같습니다.**
* API의 반환값은 객체만 던지고, 예외가 발생하면 controllerAdvice에서 핸들링하는 방식으로 구현될 것같은데 이렇게 된다면 201이나 204같은 상태코드를 반환하지 못하고 200의 상태코드만 반환되는것 처럼 보입니다.
  * **위에 말했던 commonApiResponse 객체에 해당 기능을 하는 필드를 두면 되겠습니다. 간단히 생각해봤을때. HttpStatus 객체를 가질 수 있는 필드가 필요 할 것 같습니다. (그게 실제로 response json 에 보이든 안보이든) 해당 필드를 controllerAdvice 에서 적절히 응용할 수 있을 것 같습니다. (추가적으로 http status code 말고도 세부적인 code 핸들링에 니즈가 있다면 그것도 추가하면 되겠습니다.)**
* 다른 2xx대의 상태코드를 사용하지않고 성공시에는 200의 상태코드만 반환하고 예외만 핸들링하여 따로 처리해주는 방식이 더 좋은지 궁금합니다.
  * **위 리뷰로 갈음 할 수 있는 내용으로 보입니다.**

Reply:
제가 얼추 생각한 내용을 예제로 잘 알려준것 같은데요. 구체적인건 쓰레드 참고 부탁드립니다.
* https://chatgpt.com/share/ce4808bc-c56f-4c83-8b0e-d99c78aa8624